lib/pushmi_pullyu/aip_creator.rb

```diff
@@ -0,0 +1,73 @@
+require 'pushmi_pullyu'
+require 'pushmi_pullyu/aip_downloader'
+require 'bagit'
+require 'archive/tar/minitar'
+
+class PushmiPullyu::AipCreator
+
+  attr_reader :noid, :config, :logger,
+              :workdir, :basedir, :tar_filename
+
+  def initialize(noid, config = nil, logger = nil)
+    @noid = noid
+
+    @config = config || PushmiPullyu.options
+    @logger = logger || PushmiPullyu.logger
```

Comment:
It makes understanding the non-test code path (ie, what actually happens in production) much more complicated, though. When every class and method get a logger argument passed into it, and has logger instance variables stored in it, you create a bunch of uncertainty as to what the actual object instance receiving all of these calls is. It's also inconsistent with the style being used elsewhere in this codebase.

The singletons offer an easily mockable interface for testing purposes. Adding runtime indirection to production just complicates the abstraction. significantly. Ditch it.